tools/autograd/gen_variable_type.py

```diff
@@ -828,6 +885,66 @@ def emit_increment_version(f: NativeFunction) -> List[str]:
             return []
         return [f'increment_version({r});' for r in cpp.return_names(f)]
 
+    def emit_fw_derivatives() -> List[str]:
+        content: List[str] = []
+        for derivative in fw_derivatives:
+            res = derivative.var_name
+
+            if_stmt = " || ".join([FW_DERIVATIVE_CHECK_TEMPLATE.substitute(req_inp=inp.name)
+                                   for inp in differentiable_inputs if inp.name in derivative.required_inputs_fw_grad])
+            if not if_stmt:
+                # Handle functions like stack
+                # For these, we don't unpack anything and always call the user function
+                assert len(differentiable_inputs) == 1
+                assert is_tensor_list_type(differentiable_inputs[0].type)
+                if_stmt = "true"
+            fw_grad_defined = ""
+            for inp in differentiable_inputs:
+                if inp.name in derivative.required_inputs_fw_grad:
+                    if inp.cpp_type == 'Tensor &':
+                        warnings.warn(f'The formula for "{name}" might be using the original value of {inp.name} that is being '
+                                      'modified inplace. This can lead to wrong forward gradients.')
```

Comment:
No one is ever going to read this warning. Make it an error and give people a way to explicitly squash the error.

Reply:
This is fixed in the next PR in the stack